halo2/src/prover.rs

```diff
@@ -137,11 +168,82 @@ impl<'a, F: FieldElement> Halo2Prover<'a, F> {
         Ok(proof)
     }
 
-    pub fn prove_aggr(
+    /// Generate a single proof for a given PIL using Keccak transcripts.
+    /// These proofs can be verified directly on Ethereum.
+    pub fn prove_snark_single(
```

Comment:
There seems to be a lot of duplicated code with `prove_poseidon`. The only difference is the transcript. Maybe we could have a `prove<T>()` function that is called in both cases?

Reply:
yep, that was on my TODO list. Just wanted to separate the interfaces first and then clean up the functions